### ISSUE TITLE
RLVR example训练无收益

### ISSUE THREAD

**Issue by qilong-zhang** [alibaba/ROLL#302]
hi，感谢开源ROLL这样优秀的工作～
我最近在跑demo，发现训练几乎没有效果，于是我又尝试了一下代码中默认的exampel，发现也是一样，能辛苦帮忙看看吗～

样例配置路径：ROLL/examples/qwen3-8B-rlvr_megatron/rlvr_config.yaml
启动脚本如下
```
export PYTHONPATH=$(pwd):$PYTHONPATH
export ROLL_LOG_LEVEL=DEBUG                                                                                                                                                   
# 启动训练
python examples/start_rlvr_pipeline.py \
    --config_path qwen3-8B-rlvr_megatron \
    --config_name rlvr_config \
    "$@" | tee train.log
```
yaml简单修改只做math训练，训练模型qwen3-8B，如下

```
hydra:
  run:
    dir: .
  output_subdir: null

exp_name: "qwen3-14B-rlvr-config"
seed: 42
logging_dir: {exp_name}
output_dir: {exp_name}
system_envs:
  USE_MODELSCOPE: '1'


checkpoint_config:
  type: file_system
  output_dir: ${output_dir}

# track_with: ml_tracker
# tracker_kwargs:
#  project: roll_qwen3_8B
#  notes: roll_qwen3_8B
#  tags:
#    - rlvr
#    - baseline

track_with: tensorboard
tracker_kwargs:
  log_dir: ${output_dir}

num_gpus_per_node: 4

max_steps: 500
save_steps: 100
logging_steps: 1
eval_steps: 10
resume_from_checkpoint: false


rollout_batch_size: 32  # prompt
prompt_length: 2048
response_length: 4096

num_return_sequences_in_group: 8
ppo_epochs: 1
adv_estimator: "reinforce"

# clip
value_clip: 0.5
reward_clip: 10
advantage_clip: 2.0
dual_clip_loss: true

# normalize
norm_mean_type: ~
norm_std_type: ~

# data mask
max_len_mask: true
difficulty_mask: true
difficulty_low_threshold: 0.1
difficulty_high_threshold: 0.95
error_max_len_clip: false

# data weight
difficulty_loss_weight: false
length_loss_weight: false

# reward
add_token_level_kl: false

# advantage
whiten_advantages: true

# dynamic sampling scheduler
# use_additional_prompts: true
# max_running_requests: 256
# is_num_return_sequences_expand: false

pretrain: /home/zhangqilong.zql/ROLL/model/qwen3_8b
reward_pretrain: /home/zhangqilong.zql/ROLL/model/qwen3_8b

validation:
  data_args:
    template: qwen3
    file_name:
      - data/math_benchmarks.jsonl
  generating_args:
    max_new_tokens: ${response_length}
    top_p: 0.6
    top_k: 50
    num_beams: 1
    temperature: 0.6
    num_return_sequences: 1


actor_train:
  model_args:
    disable_gradient_checkpointing: false
    dtype: bf16
    model_type: ~
  training_args:
    learning_rate: 1.0e-6
    weight_decay: 0
    per_device_train_batch_size: 16
    gradient_accumulation_steps: 4
    warmup_steps: 5
    num_train_epochs: 50
  data_args:
    template: qwen3
    file_name:
      - data/math_deepmath_deal.jsonl
    domain_interleave_probs:
      math_rule: 1
    dataset_dir: data
    messages: messages
    interleave_probs: "1.0"
    preprocessing_num_workers: 16
  strategy_args:
    strategy_name: megatron_train
    strategy_config:
      tensor_model_parallel_size: 4
      pipeline_model_parallel_size: 1
      expert_model_parallel_size: 1
      use_distributed_optimizer: true
      recompute_granularity: full
  device_mapping: list(range(0,4))
  infer_batch_size: 4

actor_infer:
  model_args:
    disable_gradient_checkpointing: true
    dtype: bf16
  generating_args:
    max_new_tokens: ${response_length}
    top_p: 0.99
    top_k: 100
    num_beams: 1
    temperature: 0.99
    num_return_sequences: ${num_return_sequences_in_group}
  data_args:
    template: qwen3
  strategy_args:
    strategy_name: vllm
    strategy_config:
      gpu_memory_utilization: 0.8
      block_size: 16
      max_model_len: 8000
  device_mapping: list(range(0,4))
  infer_batch_size: 4

reference:
  model_args:
    disable_gradient_checkpointing: true
    dtype: bf16
    model_type: ~
  data_args:
    template: qwen3
  strategy_args:
    strategy_name: megatron_infer
    strategy_config:
      tensor_model_parallel_size: 4
      pipeline_model_parallel_size: 1
      expert_model_parallel_size: 1
  device_mapping: list(range(0,4))
  infer_batch_size: 4

rewards:
  math_rule:
    worker_cls: roll.pipeline.rlvr.rewards.math_rule_reward_worker.MathRuleRewardWorker
    model_args:
      model_name_or_path: ${reward_pretrain}
    data_args:
      template: qwen3
    tag_included: [deepmath_103k, aime]
    world_size: 8
    infer_batch_size: 1
```

但是训练了50多步，reward并没有变化，score也是

<img width="2564" height="822" alt="Image" src="https://github.com/user-attachments/assets/ad07dde4-7d60-46df-a72e-a6c6dcda918d" />

<img width="2588" height="828" alt="Image" src="https://github.com/user-attachments/assets/dea68cfc-31b3-498d-a81a-c197d11329ca" />

<img width="2570" height="762" alt="Image" src="https://github.com/user-attachments/assets/56a2ff54-febf-4cd4-9e5c-6836a442a997" />

<img width="2616" height="794" alt="Image" src="https://github.com/user-attachments/assets/79dce934-26f2-4a78-a740-7c3f3d1893be" />

**Comment by wwxFromTju** [comment]
针对这个现象，可能response_length: 4096偏短，可以看一下截断的比例。或者再训一段时间看看。

另外，Qwen3-8B这个config应该是个示例config。

**Comment by qilong-zhang** [comment]
@wwxFromTju 感谢回复，我多训几步后发现score确实上涨了，但是rewards指标还是波动状态。
请问在outcome reward model场景下 需要看rewards指标吗？

**Comment by PanAndy** [comment]
我们一般主要看scores, reward 对score可能做后处理，比如算norm之类的。

**Comment by qilong-zhang** [comment]
好的 我知道啦 感谢解答～
